### PR TITLE
(WIP) Revise README.md for clarity

### PR DESCRIPTION
**Pull request by jawnsy**
Closes: #7568

Signed-off-by: Jonathan Yu jawnsy@redhat.com

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,17 +5,26 @@ OpenShift Application Platform
 [![Travis](https://travis-ci.org/openshift/origin.svg?branch=master)](https://travis-ci.org/openshift/origin)
 [![Jenkins](https://ci.openshift.redhat.com/jenkins/buildStatus/icon?job=devenv_ami)](https://ci.openshift.redhat.com/jenkins/job/devenv_ami/)
 
-OpenShift Origin is a distribution of [Kubernetes](https://kubernetes.io) optimized for continuous application development and multi-tenant deployment.  Origin adds developer and operations-centric tools on top of Kubernetes to enable rapid application development, easy deployment and scaling, and long-term lifecycle maintenance for small and large teams.
+***OpenShift Origin*** is a distribution of [Kubernetes](https://kubernetes.io) optimized for continuous application development and multi-tenant deployment.  Origin enables teams of all sizes to quickly develop, deploy, and scale applications, reducing the length of development cycles and operational effort.
 
-**Features:**
+Features
```
Comment (jawnsy) [inline README.md]: Split "Features" into two sections, one aimed at developers and the other at operational staff (sysadmins)

File: README.md [hunk 2]
```diff
@@ -23,28 +32,26 @@ OpenShift Origin is a distribution of [Kubernetes](https://kubernetes.io) optimi
 **Learn More:**
 
 * **[Public Documentation](http://docs.openshift.org/latest/welcome/index.html)**
-* The **[Trello Roadmap](https://ci.openshift.redhat.com/roadmap_overview.html)** covers the epics and stories being worked on (click through to individual items)
+  * **[API Documentation](http://docs.openshift.org/latest/rest_api/openshift_v1.html)**
 * **[Technical Architecture Presentation](https://docs.google.com/presentation/d/1Isp5UeQZTo3gh6e59FMYmMs_V9QIQeBelmbyHIJ1H_g/pub?start=false&loop=false&delayms=3000)**
 * **[System Architecture](https://github.com/openshift/openshift-pep/blob/master/openshift-pep-013-openshift-3.md)** design document
-* **[API Documentation](http://docs.openshift.org/latest/rest_api/openshift_v1.html)**
+* The **[Trello Roadmap](https://ci.openshift.redhat.com/roadmap_overview.html)** shows work (epics and user stories) that are planned or currently underway (click through to individual items)
```
Comment (jawnsy) [inline README.md]: Moved the Trello Roadmap down, because I think the docs are most relevant for users

File: README.md [hunk 3]
```diff
@@ -23,28 +32,26 @@ OpenShift Origin is a distribution of [Kubernetes](https://kubernetes.io) optimi
...
 ### Installation
 
-If you have downloaded the client tools, place the included binaries in your PATH.
-
-* For a quick install of Origin, see the [Getting Started Install guide](https://docs.openshift.org/latest/getting_started/administrators.html).
-* For an advanced installation using [Ansible](https://github.com/openshift/openshift-ansible), follow the [Advanced Installation guide](https://docs.openshift.org/latest/install_config/install/advanced_install.html)
+* If you intend to develop applications to run on an existing installation of the OpenShift platform, you can download the client tools and place the included binaries in your `PATH`
+* For local development/test or product evaluation purposes, we recommend using the quick install as described in the [Getting Started Install guide](https://docs.openshift.org/latest/getting_started/administrators.html).
+* For production environments, we recommend using [Ansible](https://github.com/openshift/openshift-ansible) as described in the [Advanced Installation guide](https://docs.openshift.org/latest/install_config/install/advanced_install.html)
 * To build and run from source, see [CONTRIBUTING.adoc](CONTRIBUTING.adoc)
```
Comment (jawnsy) [inline README.md]: Tried to better differentiate between different use cases here:

1. developers (develop locally, use remote OpenShift server)
2. local dev/test environments (quick install)
3. production environments (Ansible-based install)
4. Origin contributors (build from source)

File: README.md [hunk 4]
```diff
@@ -52,32 +59,32 @@ The Origin API is located on each server at `https://<host>:8443/oapi/v1`. These
 
 ### Kubernetes
 
-If you're looking for more information about using Kubernetes or the lower level concepts that Origin depends on, see the following:
+Since OpenShift Origin builds on Kubernetes, it is helpful to understand underlying concepts such as Pods and Replication Controllers. The following are good references:
 
-* [Kubernetes Getting Started](https://github.com/kubernetes/kubernetes/blob/master/README.md)
+* [Kubernetes User Guide](http://kubernetes.io/docs/user-guide/)
```
Comment (jawnsy) [inline README.md]: Updated links to Kubernetes User Guide as referenced by the corresponding README in GitHub

**Comment by kargakis** [comment]
I like it :+1:

**Comment by jawnsy** [comment]
@smarterclayton @danmcp I think this is about ready for a merge. Can one of you please take a look, or nominate someone else?

I tried to add some annotations explaining the changes I made. Some of it was just rephrasing and reformatting. I apologize for the diffs, they're a bit difficult to read as a result.

File: README.md [hunk 5]
```diff
@@ -5,79 +5,86 @@ OpenShift Application Platform
...
+* Easily build applications with integrated service discovery and persistent storage
+* Quickly and easily scale applications to handle periods of increased demand
+  * Support for automatic high availability, load-balancing, health checking, and fail-over
+* Push source code to your Git repository and automatically deploy containerized applications
+* Web console and command-line client for building and monitoring applications
+
+**For system administrators:**
 
-* Easily build applications with integrated service discovery, DNS, load balancing, failover, health checking, persistent storage, and fast scaling
-* Push source code to your Git repository and automatically roll out containerized applications (micro or macro)
```
Comment (Miciah) [inline README.md]: You dropped "micro or macro", which was an already understated reference to the goal of supporting both microservice-based and traditional monolithic applications on the platform.  Is that change intentional?

File: README.md [hunk 6]
```diff
@@ -5,79 +5,86 @@ OpenShift Application Platform
...
-* Easy to use client and web console for building and monitoring applications
 * Centralized administration and management of an entire stack, team, or organization
-  * Templatize the components of your system, reuse them, and iteratively deploy them over time
-  * Roll out changes to software stacks to your entire organization in a controlled fashion
-* Team and user isolation of containers, builds, and network communication in an easy multi-tenant system
+  * Create reusable templates for components of your system, and iteratively deploy them over time
+  * Roll out modifications to software stacks to your entire organization in a controlled fashion
+  * Integration with your existing authentication mechanisms, including LDAP, Active Directory, and public OAuth providers such as GitHub
+* Multi-tenancy support, including team and user isolation of containers, builds and network communication
```
Comment (Miciah) [inline README.md]: We include the penultimate comma elsewhere—consistency is good.

File: README.md [hunk 7]
```diff
@@ -5,79 +5,86 @@ OpenShift Application Platform
...
 
 ### Installation
 
-If you have downloaded the client tools, place the included binaries in your PATH.
-
-* For a quick install of Origin, see the [Getting Started Install guide](https://docs.openshift.org/latest/getting_started/administrators.html).
-* For an advanced installation using [Ansible](https://github.com/openshift/openshift-ansible), follow the [Advanced Installation guide](https://docs.openshift.org/latest/install_config/install/advanced_install.html)
+* If you intend to develop applications to run on an existing installation of the OpenShift platform, you can download the client tools and place the included binaries in your `PATH`
+* For local development/test or product evaluation purposes, we recommend using the quick install as described in the [Getting Started Install guide](https://docs.openshift.org/latest/getting_started/administrators.html).
+* For production environments, we recommend using [Ansible](https://github.com/openshift/openshift-ansible) as described in the [Advanced Installation guide](https://docs.openshift.org/latest/install_config/install/advanced_install.html)
```
Comment (Miciah) [inline README.md]: We should be consistent in use of the fullstop at the end of each bulletpoint item.

File: README.md [hunk 8]
```diff
@@ -89,11 +96,21 @@ FAQ
 
 3. What can I run on Origin?
 
-    Origin is designed to run any existing Docker images.  In addition you can define builds that will produce new Docker images from a Dockerfile.  However the real magic of Origin can be seen when using [Source-To-Image](https://github.com/openshift/source-to-image) builds which allow you to simply supply an application source repository which will be combined with an existing Source-To-Image enabled Docker image to produce a new runnable image that runs your application.  We are continuing to grow the ecosystem of Source-To-Image enabled images and documenting them [here](http://docs.openshift.org/latest/using_images/s2i_images/overview.html). Our available images are:
+    Origin is designed to run any existing Docker images.  Additionally, you can define builds
+    that will produce new Docker images using a `Dockerfile`.
+
+    However, the real magic of Origin is [Source-to-Image (S2I)](https://github.com/openshift/source-to-image)
```
Comment (Miciah) [inline README.md]: I think there's more too Origin than S2I.  *grin*.  Maybe "the real magic of Origin can be illustrated by" or similar?

File: README.md [hunk 9]
```diff
@@ -89,11 +96,21 @@ FAQ
...
+    builds, which allow developers to simply provide an application source repository containing
+    code to build and execute.  It works by combining an existing S2I-enabled Docker image with
+    application source to produce a new runnable image for your application.
+    
+    We are continuing to grow the [ecosystem of Source-to-Image builder images](http://docs.openshift.org/latest/using_images/s2i_images/overview.html)
```
Comment (Miciah) [inline README.md]: Might as well re-use the "S2I" abbreviation since you defined it above.

File: README.md [hunk 10]
```diff
@@ -111,39 +128,63 @@ FAQ
...
-      * These settings are per project, so containers in different projects cannot see each other by default
+      * Any persistent storage they access will be under a unique SELinux label, which
+        prevents others from seeing their content
+      * These settings are per project, so containers in different projects cannot see
+        each other by default
     * Regular users can run Docker, source, and custom builds
-      * By default, Docker builds can (and often do) run as root. You can control who can create Docker builds through the `builds/docker` and `builds/custom` policy resource.
+      * By default, Docker builds can (and often do) run as root. You can control who
+        can create Docker builds through the `builds/docker` and `builds/custom` policy
+        resource.
```
Comment (Miciah) [inline README.md]: Again, the use of fullstops in list items is inconsistent.  (I prefer to use a fullstop always after a full sentence.)

File: README.md [hunk 11]
```diff
@@ -111,39 +128,63 @@ FAQ
...
 
 Contributing
 ------------
 
-You can develop [locally on your host](CONTRIBUTING.adoc#develop-locally-on-your-host) or with a [virtual machine](CONTRIBUTING.adoc#develop-on-virtual-machine-using-vagrant), or if you want to just try out Origin [download the latest Linux server, or Windows and Mac OS X client pre-built binaries](CONTRIBUTING.adoc#download-from-github).
+You can develop [locally on your host](CONTRIBUTING.adoc#develop-locally-on-your-host) or with
+a [virtual machine](CONTRIBUTING.adoc#develop-on-virtual-machine-using-vagrant).
+
+If you just want to try Origin, [download the latest pre-built binaries](CONTRIBUTING.adoc#download-from-github)
+for Linux, MacOS X (client only), or Windows (client only.)
```
Comment (Miciah) [inline README.md]: Whoops, the parentheses swallowed the fullstop!

File: README.md (hunk 5 above)
Comment (jawnsy) [inline README.md]: I've been pondering this one, and I'm not sure if it's worth mentioning. I think it's useful that we can run both stateful and stateless apps, but I'm not sure it's valuable to explicitly call out microservice/monolithic applications as the distinction isn't that important. For microservices, you need service discovery (to find your other pieces), and for most older apps, you generally need persistent storage (though that could be true of microservices too - consider a store like etcd or mysql)

**Comment by jawnsy** [comment]
Hey Michail @kargakis, can you PTAL? I noticed you have merge permissions :)

**Comment by kargakis** [comment]
Changes in the README and other user-facing docs are mostly @danmcp's area

File: README.md [hunk 12]
```diff
@@ -6,82 +6,82 @@ OpenShift Application Platform
...
 [![Jenkins](https://ci.openshift.redhat.com/jenkins/buildStatus/icon?job=devenv_ami)](https://ci.openshift.redhat.com/jenkins/job/devenv_ami/)
 [![Join the chat at freenode:openshift-dev](https://img.shields.io/badge/irc-freenode%3A%20%23openshift--dev-blue.svg)](http://webchat.freenode.net/?channels=%23openshift-dev)
+[![Licensed under Apache License version 2.0](https://img.shields.io/github/license/openshift/origin.svg?maxAge=2592000)](https://www.apache.org/licenses/LICENSE-2.0)
 
-OpenShift Origin is a distribution of [Kubernetes](https://kubernetes.io) optimized for continuous application development and multi-tenant deployment.  Origin adds developer and operations-centric tools on top of Kubernetes to enable rapid application development, easy deployment and scaling, and long-term lifecycle maintenance for small and large teams.
+***OpenShift Origin*** is a distribution of [Kubernetes](https://kubernetes.io) optimized for continuous application development and multi-tenant deployment.  Origin enables teams of all sizes to quickly develop, deploy, and scale applications, reducing the length of development cycles and operational effort.
```
Comment (danmcp) [inline README.md]: The change to this sentence takes away the info that we are additive to kube and don't just use kube.

File: README.md [hunk 13]
```diff
@@ -6,82 +6,82 @@ OpenShift Application Platform
...
-* Push source code to your Git repository and automatically roll out containerized applications (micro or macro)
-* Easy to use client and web console for building and monitoring applications
-* Centralized administration and management of an entire stack, team, or organization
-  * Templatize the components of your system, reuse them, and iteratively deploy them over time
-  * Roll out changes to software stacks to your entire organization in a controlled fashion
-* Team and user isolation of containers, builds, and network communication in an easy multi-tenant system
-  * Allow developers to run containers securely with fine-grained controls in production
-  * Limit, track, and manage the developers and teams on the platform
-* Integrated Docker registry, automatic edge load balancing, cluster logging, and integrated metrics
+**For developers:**
```
Comment (danmcp) [inline README.md]: Creating this split somewhat takes away from the devops mindset.  It's valid for some people but not for others to do so.

File: README.md [hunk 14]
```diff
@@ -6,82 +6,82 @@ OpenShift Application Platform
...
 ---------------
 
 ### Installation
 
-If you have downloaded the client tools, place the included binaries in your PATH.
-
-* For a quick install of Origin, see the [Getting Started Install guide](https://docs.openshift.org/latest/getting_started/administrators.html).
-* For an advanced installation using [Ansible](https://github.com/openshift/openshift-ansible), follow the [Advanced Installation guide](https://docs.openshift.org/latest/install_config/install/advanced_install.html)
-* To build and run from source, see [CONTRIBUTING.adoc](CONTRIBUTING.adoc)
+* If you intend to develop applications to run on an existing installation of the OpenShift platform, you can download the client tools and place the included binaries in your `PATH`.
```
Comment (danmcp) [inline README.md]: Where is the link to download the client tools?

File: README.md (hunk 13 above)
Comment (smarterclayton) [inline README.md]: Before this merges I want to give it another look as well.

File: README.md [hunk 15]
```diff
@@ -6,82 +6,82 @@ OpenShift Application Platform
...
+* For production environments, we recommend using [Ansible](https://github.com/openshift/openshift-ansible) as described in the [Advanced Installation guide](https://docs.openshift.org/latest/install_config/install/advanced_install.html).
+* To build and run from source, see [CONTRIBUTING.adoc](CONTRIBUTING.adoc).
 
 ### Concepts
 
-We highly recommend trying out the [Origin walkthrough](https://github.com/openshift/origin/blob/master/examples/sample-app/README.md) which covers the core concepts in Origin.  The walkthrough is accompanied by a blog series on [blog.openshift.com](https://blog.openshift.com/openshift-v3-deep-dive-docker-kubernetes/) that goes into more detail.  It's a great place to start.
+The [Origin walkthrough](https://github.com/openshift/origin/blob/master/examples/sample-app/README.md) is a step-by-step guide that demonstrates the core capabilities of OpenShift throughout the development, build, and test cycle.  The walkthrough is accompanied by a blog series on [blog.openshift.com](https://blog.openshift.com/openshift-v3-deep-dive-docker-kubernetes/) that goes into more detail.  It's a great place to start.
```
Comment (danmcp) [inline README.md]: build, deploy, and

File: README.md [hunk 16]
```diff
@@ -6,82 +6,82 @@ OpenShift Application Platform
...
 
 ### Origin API
 
-The Origin API is located on each server at `https://<host>:8443/oapi/v1`. These APIs are described via [Swagger v1.2](https://www.swagger.io) at `https://<host>:8443/swaggerapi/oapi/v1`. For more, [see the API documentation](http://docs.openshift.org/latest/rest_api/openshift_v1.html).
+The Origin API is located on each server at `https://<host>:8443/oapi/v1`. These APIs are described via [Swagger v1.2](https://www.swagger.io) at `https://<host>:8443/swaggerapi/oapi/v1`. For more, [see the API documentation](https://docs.openshift.org/latest/rest_api/openshift_v1.html).
 
 ### Kubernetes
 
-If you're looking for more information about using Kubernetes or the lower level concepts that Origin depends on, see the following:
+Since OpenShift Origin builds on Kubernetes, it is helpful to understand underlying concepts such as Pods and Replication Controllers. The following are good references:
```
Comment (danmcp) [inline README.md]: Not sure this is any better.

File: README.md (hunk 12 above)
Comment (jawnsy) [inline README.md]: @danmcp Thanks for the review. "OpenShift Origin is a distribution of Kubernetes" isn't sufficient to indicate that we extend Kube? Would it be clearer to rephrase as:

> Origin extends Kubernetes, enabling teams of all sizes to quickly develop, deploy, and scale applications, reducing the length of development cycles and operational effort.

I just worry that this makes the sentence too complicated, since we're trying to convey three separate things:

1. that we're extending Kube;
2. that we have features to make development/deployment/scaling easier;
3. that we anticipate benefits like faster time-to-market/quicker development cycles, and ease of operation

Or do you have another phrasing in mind that might be clearer? I'm just trying to avoid being too redundant and buzzwordy; the first sentence is our chance to articulate our value to users.

File: README.md [hunk 17]
```diff
@@ -6,82 +6,82 @@ OpenShift Application Platform
...
 ---
 
 1. How does Origin relate to Kubernetes?
 
-    Origin is a distribution of Kubernetes optimized for enterprise application development and deployment,
-    used by OpenShift 3 and Atomic Enterprise.  Origin embeds Kubernetes and adds additional
-    functionality to offer a simple, powerful, and easy-to-approach developer and operator experience
-    for building applications in containers.  Our goal is to do most of that work upstream, with
-    integration and final packaging occurring in Origin.
+    Origin is a distribution of Kubernetes optimized for enterprise application development and deployment, and is the foundation of OpenShift 3 and Atomic Enterprise.  Origin extends Kubernetes with additional functionality, offering a simple, yet powerful, development and operational experience.  Both Origin and the upstream Kubernetes project focus on building and deploying applications in containers.
```
Comment (danmcp) [inline README.md]: Remove '3 and Atomic Enterprise'.

Kubernetes doesn't have builds.

Not sure the overall goal of changing the last sentence.  It has a very different meaning.

File: README.md (hunk 13 above)
Comment (jawnsy) [inline README.md]: @danmcp agreed, but isn't it easier for a DevOps minded person to read both sections than it is for a team with split Dev and Ops responsibilities to individually pick out the parts that are relevant to them?

File: README.md [hunk 18]
```diff
@@ -93,16 +93,20 @@ FAQ
...
+    We are continuing to grow the [ecosystem of Source-to-Image builder images](https://docs.openshift.org/latest/using_images/s2i_images/overview.html) and it's straightforward to [create your own](https://blog.openshift.com/create-s2i-builder-image/).  Our available images are:
 
     * [Ruby](https://github.com/openshift/sti-ruby)
     * [Python](https://github.com/openshift/sti-python)
-    * [NodeJS](https://github.com/openshift/sti-nodejs)
+    * [Node.js](https://github.com/openshift/sti-nodejs)
     * [PHP](https://github.com/openshift/sti-php)
     * [Perl](https://github.com/openshift/sti-perl)
-    * [Wildfly](https://github.com/openshift/wildfly-8-centos)
+    * Other community-supported images, including [WildFly](https://github.com/openshift-s2i/s2i-wildfly)
```
Comment (danmcp) [inline README.md]: Why reference other images here and only have a link to wildfly?

File: README.md (hunk 14 above)
Comment (jawnsy) [inline README.md]: @danmcp What's the best link to use? Would that be the [GitHub releases page](https://github.com/openshift/origin/releases)?

File: README.md (hunk 18 above)
Comment (jawnsy) [inline README.md]: @danmcp I just wanted to clarify that WildFly is a community image, not an "official" one. Is that a worthwhile distinction for the readme, or is it better to revert to the way it was before? (I'd still update the WildFly name and the URL, though)

File: README.md [hunk 19]
```diff
@@ -112,46 +116,50 @@ FAQ
...
+    Although we recommend against it for security reasons, it is also possible to relax these restrictions as described in the [security documentation](https://docs.openshift.org/latest/admin_guide/manage_scc.html).
 
 5. How do I get networking working?
 
-    The Origin and Kubernetes network model assigns each pod (group of containers) an IP that is expected to be reachable from all nodes in the cluster. The default setup is through a simple SDN plugin with OVS - this plugin expects the port 4679 to be open between nodes in the cluster. Also, the Origin master processes need to be able to reach pods via the network, so they may require the SDN plugin.
+    The Origin and Kubernetes network model assigns each Pod (group of containers) an IP address that is expected to be reachable from all nodes in the cluster. The default configuration uses Open vSwitch to provide Software-Defined Networking (SDN) capabilities, which requires communication between nodes in the cluster using port 4679.  Additionally, the Origin master processes must be able to reach pods within the network, so they may require the SDN plugin.
```
Comment (danmcp) [inline README.md]: add (OVS) after Open vSwitch

File: README.md (hunk 18 above)
Comment (smarterclayton) [inline README.md]: For OpenShift Origin (this repo), the distinction is without a difference
for the audience.

File: README.md (hunk 12 above)
Comment (danmcp) [inline README.md]: > OpenShift Origin is a distribution of Kubernetes" isn't sufficient to indicate that we extend Kube

I read those two sentences and it wasn't clear what origin is trying to be.  It could be nothing more than redistributing kube.

I personally still prefer the original sentence still.  The only thing that bothers me is it doesn't list build and test/ci/cd.  Although you can maybe assume those in develop.

File: README.md (hunk 13 above)
Comment (danmcp) [inline README.md]: @jawnsy Some of the items in the admin section a dev will care about though.

File: README.md (hunk 14 above)
Comment (danmcp) [inline README.md]: Yeah I think so.

File: README.md (hunk 18 above)
Comment (danmcp) [inline README.md]: All of these are only community supported right?

File: README.md [hunk 20]
```diff
@@ -112,46 +116,50 @@ FAQ
...
 
     Other networking options are available such as Calico, Flannel, Nuage, and Weave. For a non-overlay networking solution, existing networks can be used by assigning a different subnet to each host, and ensuring routing rules deliver packets bound for that subnet to the host it belongs to. This is called [host subnet routing](https://docs.openshift.org/latest/admin_guide/native_container_routing.html).
```
Comment (jawnsy) [inline README.md]: @danmcp Wondering if we should add links to the Calico, Flannel, Nuage, and Weave project sites here, or if those options are even recommended for use with OpenShift. Do we provide guides for using those options with OpenShift? If so, maybe we should link to those?

File: README.md (hunk 13 above)
Comment (jawnsy) [inline README.md]: @danmcp I guess my concern is that just listing a long list of features will hurt readability and make the file less approachable. The structure is meant to imply "as a (Developer | Operator), you're probably most interested in X, Y, Z" but doesn't mean that both groups shouldn't read through everything. It lends a little context as to why the features are worthwhile, and for a DevOps person, informs which "hat" you should wear while reading through each point.

If you feel strongly about this, I will merge the two sections together. I feel that we're doing our users a disservice by just providing a laundry list of features without connection to the specific personas that they target.

File: README.md [hunk 21]
```diff
@@ -6,82 +6,82 @@ OpenShift Application Platform
...
 
-OpenShift Origin is a distribution of [Kubernetes](https://kubernetes.io) optimized for continuous application development and multi-tenant deployment.  Origin adds developer and operations-centric tools on top of Kubernetes to enable rapid application development, easy deployment and scaling, and long-term lifecycle maintenance for small and large teams.
+***OpenShift Origin*** is a distribution of [Kubernetes](https://kubernetes.io) optimized for continuous application development and multi-tenant deployment.  Origin enables teams of all sizes to quickly develop, deploy, and scale applications, reducing the length of development cycles and operational effort.
 
 [![Watch the full asciicast](docs/openshift-intro.gif)](https://asciinema.org/a/49402)
 
 **Features:**
 
-* Easily build applications with integrated service discovery, DNS, load balancing, failover, health checking, persistent storage, and fast scaling
-* Push source code to your Git repository and automatically roll out containerized applications (micro or macro)
```
Comment (jawnsy) [inline README.md]: I dropped the "micro or macro" point as I don't think the term "macro service" is well established (most people see this as a distinction between stateful vs stateless, or microservice vs monolith). On the other hand, the ability to run legacy workloads (especially stateful applications persisting on disk) is an important differentiator for OpenShift. I'd welcome feedback on ways we can make this clearer - for now I've just tried to explicitly mention support for persistent storage.

File: README.md (hunk 13 above)
Comment (smarterclayton) [inline README.md]: There is no longer a clean separation between the personas, especially in
the audience our open source community reaches.  I don't mind ordering
giving a subtle nod, but I don't see a strong distinction in the people who
care about this readme.

On Jun 20, 2016, at 7:23 PM, Jonathan Yu <notifications@github.com> wrote:

In README.md
<https://github.com/openshift/origin/pull/8007#discussion_r67784876>:

>
>  [Watch the full asciicast](https://asciinema.org/a/49402)
>
>  **Features:**
>
> -* Easily build applications with integrated service discovery, DNS, load balancing, failover, health checking, persistent storage, and fast scaling
> -* Push source code to your Git repository and automatically roll out containerized applications (micro or macro)
> -* Easy to use client and web console for building and monitoring applications
> -* Centralized administration and management of an entire stack, team, or organization
> -  * Templatize the components of your system, reuse them, and iteratively deploy them over time
> -  * Roll out changes to software stacks to your entire organization in a controlled fashion
> -* Team and user isolation of containers, builds, and network communication in an easy multi-tenant system
> -  * Allow developers to run containers securely with fine-grained controls in production
> -  * Limit, track, and manage the developers and teams on the platform
> -* Integrated Docker registry, automatic edge load balancing, cluster logging, and integrated metrics
> +**For developers:**

@danmcp <https://github.com/danmcp> I guess my concern is that just listing
a long list of features will hurt readability and make the file less
approachable. The structure is meant to imply "as a (Developer | Operator),
you're probably most interested in X, Y, Z" but doesn't mean that both
groups shouldn't read through everything. It lends a little context as to
why the features are worthwhile, and for a DevOps person, informs which
"hat" you should wear while reading through each point.

If you feel strongly about this, I will merge the two sections together. I
feel that we're doing our users a disservice by just providing a laundry
list of features without connection to the specific personas that they
target.

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/8007/files/bc4da8fa99b5d01b492ccd85bada33fcb9cd33dd#r67784876>,
or mute the thread
<https://github.com/notifications/unsubscribe/ABG_pxp2_deWhVk7HVTboy9vVfMRfLGoks5qNyD8gaJpZM4HwhAG>
.

File: README.md [hunk 22]
```diff
@@ -6,82 +6,82 @@ OpenShift Application Platform
...
-* **[API Documentation](http://docs.openshift.org/latest/rest_api/openshift_v1.html)**
+* The **[Trello Roadmap](https://ci.openshift.redhat.com/roadmap_overview.html)** covers the epics and stories being worked on (click through to individual items).
 
 For questions or feedback, reach us on [IRC on #openshift-dev](https://botbot.me/freenode/openshift-dev/) on Freenode or post to our [mailing list](https://lists.openshift.redhat.com/openshiftmm/listinfo/dev).
 
-
 Getting Started
 ---------------
 
 ### Installation
```
Comment (jawnsy) [inline README.md]: We don't mention the `oc cluster up` method yet, should we add it? It's not one of the [documented installation options](https://docs.openshift.org/latest/getting_started/administrators.html), either. From what little I understand of it, this might be the easiest way to get an OpenShift installation running for development purposes, since it only requires `docker` and a copy of the `oc` binary.

Comment (smarterclayton) [inline README.md]: It'll be added when I rev to 1:3.0-alpha.2

On Jun 20, 2016, at 7:34 PM, Jonathan Yu <notifications@github.com> wrote:

In README.md
<https://github.com/openshift/origin/pull/8007#discussion_r67785927>:

>
>  For questions or feedback, reach us on [IRC on #openshift-dev](https://botbot.me/freenode/openshift-dev/) on Freenode or post to our [mailing list](https://lists.openshift.redhat.com/openshiftmm/listinfo/dev).
>
> -
>  Getting Started
>  ---------------
>
>  ### Installation

We don't mention the oc cluster up method yet, should we add it? It's not
one of the documented installation options
<https://docs.openshift.org/latest/getting_started/administrators.html>,
either. From what little I understand of it, this might be the easiest way
to get an OpenShift installation running for development purposes, since it
only requires docker and a copy of the oc binary.

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/8007/files/bc4da8fa99b5d01b492ccd85bada33fcb9cd33dd#r67785927>,
or mute the thread
<https://github.com/notifications/unsubscribe/ABG_p_sq7BN1UCmSrzyXc9vP7K3465dtks5qNyOVgaJpZM4HwhAG>
.

File: README.md (hunk 20 above)
Comment (danmcp) [inline README.md]: Not as of now.

File: README.md (hunk 12 above)
Comment (derekwaynecarr) [inline README.md]: I agree with Dan, but I think we could say "is a distribution of Kubernetes that layers on additional features for ..."
